pol-core/pol/module/utilmod.cpp

```diff
@@ -164,11 +224,13 @@ Bscript::BObjectImp* UtilExecutorModule::mf_StrFormatTime()
     }
   }
 
-  char buffer[102];  // +2 for the \0 termination.
-  if ( strftime( buffer, sizeof buffer, format_string->data(), &time_struct ) > 0 )
-    return new String( buffer );
-  else
-    return new BError( "Format string too long." );
+  std::string buffer;
```

Comment:
Should you use `std::vector<char>` here instead? Non-const `buffer.data()` is new in C++17 and it comes with a [warning](https://en.cppreference.com/w/cpp/string/basic_string/data):

> 2) Modifying the past-the-end null terminator stored at data()+size() to any value other than CharT() has undefined behavior.

I guess because `str.size()` does not update when you modify `str.data()` directly. Anyway, an `std::vector<char>` would work the same way and avoid UB.

Reply:
Good point, I changed it to be on the safe side.